src/workers/build/mod.rs

```diff
@@ -0,0 +1,151 @@
+mod krate;
+
+use std::fs;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use super::{Resource, Script, WasmModule, Worker};
+
+use crate::commands::build::wranglerjs;
+use crate::commands::publish::package::Package;
+use crate::settings::project::{Project, ProjectType};
+use crate::terminal::message;
+use crate::{commands, install};
+
+pub fn build(project: &Project) -> Result<Worker, failure::Error> {
+    match project.project_type {
+        ProjectType::JavaScript => {
+            message::info("JavaScript project found. Skipping unnecessary build!")
+        }
+        ProjectType::Rust => {
+            let tool_name = "wasm-pack";
+            let binary_path = install::install(tool_name, "rustwasm")?.binary(tool_name)?;
+            let args = ["build", "--target", "no-modules"];
+
+            let command = command(&args, binary_path);
+            let command_name = format!("{:?}", command);
+
+            commands::run(command, &command_name)?;
+        }
+        ProjectType::Webpack => {
+            wranglerjs::run_build(project)?;
+        }
+    }
+
+    let worker = match project.project_type {
+        ProjectType::Rust => wasm_worker(&project.name)?,
+        ProjectType::Webpack => webpack_worker(&project.name)?,
+        ProjectType::JavaScript => js_worker(&project.name)?,
+    };
+    // add other resoureces
+    Ok(worker)
+}
+
+fn wasm_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let name = krate::Krate::new("./")?.name.replace("-", "_");
+
+    build_generated_dir()?;
+    concat_js(&name)?;
+
+    let script_path = "./worker/generated/script.js";
+    let wasm_path = format!("./pkg/{}_bg.wasm", name);
+
+    let wasm_resource = wasm_resource(wasm_path)?;
+
+    Ok(Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
```

Comment:
is this ever not hardcoded? i'd prefer to make it a default than put it in the code like this

Reply:
I'll make this a default instead. This was more or less a direct port of what the previous code was doing.